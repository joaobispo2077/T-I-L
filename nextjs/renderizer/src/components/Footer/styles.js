import styled, { css } from 'styled-components';

import { Container as TextContainer } from '../Text/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
	${({ theme }) => css`
		text-align: center;
		border-top: 0.1rem solid ${theme.colors.mediumGray};
		font-size: ${theme.font.sizes.small};

		a {
			color: inherit;
			text-decoration: none;
		}

		${TextContainer} {
			font-size: ${theme.font.sizes.small};
		}

		${SectionContainer} {
			padding-top: 0;
			padding-bottom: 0;
		}
	`}
`;
