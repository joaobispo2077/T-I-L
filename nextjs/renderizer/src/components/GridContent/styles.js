import styled, { css } from 'styled-components';

export const Container = styled.div`
	${() => css`
		text-align: center;
		max-width: 58rem;
		margin: 0 auto;
	`}
`;

export const Html = styled.div`
	${({ theme }) => css`
		margin: ${theme.spacings.xhuge};
	`}
`;
