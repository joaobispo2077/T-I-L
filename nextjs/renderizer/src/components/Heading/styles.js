import styled, { css } from 'styled-components';

const getTitleSize = {
	small: (theme) => theme.font.sizes.medium,
	medium: (theme) => theme.font.sizes.large,
	big: (theme) => theme.font.sizes.xlarge,
	huge: (theme) => theme.font.sizes.xhuge,
};

const getMediaFont = (theme) => css`
	@media ${theme.media.lteMedium} {
		font-size: ${theme.font.sizes.xlarge};
	}
`;

const getTitleCase = (uppercase) => css`
	text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
	color: ${({ theme, darken }) =>
		darken ? theme.colors.dark : theme.colors.light};
	font-size: ${({ size, theme }) => getTitleSize[size](theme)};

	${({ uppercase, theme }) => css`
		${getTitleCase(uppercase)}
		${getMediaFont(theme)}
	`};
`;
