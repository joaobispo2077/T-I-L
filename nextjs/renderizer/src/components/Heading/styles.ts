import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const getTitleSize = {
  small: (theme: DefaultTheme) => theme.font.sizes.medium,
  medium: (theme: DefaultTheme) => theme.font.sizes.large,
  big: (theme: DefaultTheme) => theme.font.sizes.xlarge,
  huge: (theme: DefaultTheme) => theme.font.sizes.xhuge,
};

const getMediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const getTitleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  color: ${({ theme, darken }) =>
    darken ? theme.colors.dark : theme.colors.light};
  font-size: ${({ size, theme }) => getTitleSize[size](theme)};

  ${({ uppercase, theme }) => css`
    ${getTitleCase(uppercase)}
    ${getMediaFont(theme)}
  `};
`;
