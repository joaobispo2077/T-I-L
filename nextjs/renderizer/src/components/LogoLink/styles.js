import styled, { css } from 'styled-components';

export const Container = styled.a`
	${() => css`
		display: flex;
		align-items: center;
		text-decoration: none;
		color: inherit;

		& > img {
			height: 3rem;
		}
	`}
`;
