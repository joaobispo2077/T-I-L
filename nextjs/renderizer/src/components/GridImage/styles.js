import styled, { css } from 'styled-components';

import { Container as TextContainer } from '../Text/styles';

export const Container = styled.div`
	${({ theme }) => css`
		${TextContainer} {
			margin-bottom: ${theme.spacings.large};
		}
	`}
`;

export const Grid = styled.div`
	${({ theme }) => css`
		counter-reset: grid-counter;
		display: grid;
		grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
		gap: ${theme.spacings.large};

		@media ${theme.media.lteMedium} {
			grid-template-columns: 1fr;
		}
	`}
`;

export const GridElement = styled.div`
	${() => css`
		overflow: hidden;
	`}
`;

export const Image = styled.img`
	${() => css`
		width: 100%;
		cursor: pointer;
		transition: transform 0.25s ease-in-out;

		&:hover {
			transform: scale(1.2) rotate(10deg);
		}
	`}
`;
