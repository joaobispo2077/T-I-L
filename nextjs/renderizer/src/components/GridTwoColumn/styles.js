import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
	${({ theme }) => css`
		display: grid;
		grid-template-columns: 1fr 2fr;

		align-items: center;
		gap: ${theme.spacings.large};

		${Title} {
			margin-bottom: ${theme.spacings.large};
		}

		@media ${theme.media.lteMedium} {
			grid-template-columns: 1fr;
		}
	`}
`;

export const TextContainer = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			margin-bottom: ${theme.spacings.large};
			text-align: center;
		}
	`}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img`
	width: 100%;
`;
