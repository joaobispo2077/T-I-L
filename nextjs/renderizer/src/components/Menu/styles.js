import styled, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
	${({ theme, visible }) => css`
		position: fixed;
		z-index: 5;
		top: 0;
		right: 0;
		left: 0;

		width: 100%;

		background: ${theme.colors.light};
		border-bottom: ${theme.colors.mediumGray};
		transition: all 300ms ease-out;

		& > ${SectionContainer} {
			padding-top: 0;
			padding-bottom: 0;
		}

		& ${Heading} {
			margin-top: 0;
			margin-bottom: 0;
		}

		@media ${theme.media.lteMedium} {
			height: 100vh;
			visibility: ${visible ? 'visible' : 'hidden'};
			opacity: ${visible ? 1 : 0};

			& > ${SectionContainer} {
				display: grid;
				grid-template-columns: 1fr;
				grid-template-rows: 1fr;

				height: 100vh;
				align-items: center;

				overflow-y: auto;
			}

			& ${Heading} {
				display: flex;
				justify-content: center;
				padding-bottom: ${theme.spacings.large};
			}
		}
	`}
`;

export const MenuContainer = styled.div`
	${({ theme }) => css`
		display: flex;
		justify-content: space-between;
		align-items: center;

		@media ${theme.media.lteMedium} {
			display: block;
			text-align: center;
			padding: ${theme.spacings.xxlarge};
		}
	`}
`;

export const Button = styled.button`
	${({ theme }) => css`
		z-index: 6;
		position: fixed;
		top: 2rem;
		right: 2rem;
		width: 4rem;
		height: 4rem;
		background: ${theme.colors.primary};
		color: ${theme.colors.light};
		cursor: pointer;
		border: none;
		display: none;

		@media ${theme.media.lteMedium} {
			display: flex;
			align-items: center;
			justify-content: center;
		}

		> svg {
			width: 2.5rem;
			height: 2.5rem;
		}
	`}
`;
