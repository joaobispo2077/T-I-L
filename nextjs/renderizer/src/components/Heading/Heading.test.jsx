import { render, screen } from '@testing-library/react';
import { Heading } from '.';
import { renderTheme } from '../../styles/themes/renderTheme';
import { themes } from '../../styles/themes';

describe('<Heading />', () => {
	it('should render with default values', () => {
		render(renderTheme(<Heading>texto</Heading>));
		expect.assertions(2);

		const heading = screen.getByRole('heading', { name: 'texto' });
		expect(heading).toBeInTheDocument();
		expect(heading).toHaveStyle({
			color: themes['default'].colors.primary,
			'font-size': themes['default'].font.sizes.xhuge,
			'text-transform': 'none',
		});
	});

	it('should render with darken color', () => {
		render(renderTheme(<Heading darken>texto</Heading>));

		const heading = screen.getByRole('heading', { name: 'texto' });
		expect(heading).toHaveStyle({
			color: themes['default'].colors.dark,
		});
	});

	it('should render sizes correctly', () => {
		const { rerender } = render(
			renderTheme(<Heading size="small">texto</Heading>),
		);

		const heading = screen.getByRole('heading', { name: 'texto' });
		expect(heading).toHaveStyle({
			'font-size': themes['default'].font.sizes.medium,
		});

		rerender(renderTheme(<Heading size="medium">texto</Heading>));

		expect(heading).toHaveStyle({
			'font-size': themes['default'].font.sizes.large,
		});

		rerender(renderTheme(<Heading size="big">texto</Heading>));

		expect(heading).toHaveStyle({
			'font-size': themes['default'].font.sizes.xlarge,
		});
	});

	it('should adapt font-size when screen is less than 768px', () => {
		render(renderTheme(<Heading size="huge">texto</Heading>));

		expect(screen.getByRole('heading', { name: 'texto' })).toHaveStyleRule(
			'font-size',
			themes['default'].font.sizes.xlarge,
			{
				media: themes['default'].media.lteMedium,
			},
		);
	});

	it('should render uppercase text', () => {
		render(renderTheme(<Heading uppercase>texto</Heading>));

		const heading = screen.getByRole('heading', { name: 'texto' });
		expect(heading).toHaveStyle({
			'text-transform': 'uppercase',
		});
	});

	it('should render correct element', () => {
		const { container } = render(renderTheme(<Heading as="h6">texto</Heading>));

		const headingSix = container.querySelector('h6');

		expect(headingSix.tagName.toLowerCase()).toBe('h6');
	});

	it('should match with snapshot', () => {
		const { container } = render(renderTheme(<Heading as="h6">texto</Heading>));

		expect(container.firstChild).toMatchSnapshot();
	});
});
