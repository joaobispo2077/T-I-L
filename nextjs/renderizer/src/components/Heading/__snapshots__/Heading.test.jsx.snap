// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match with snapshot 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h6
  class="c0"
>
  texto
</h6>
`;
